File: 6_ZigZagConversion/6_v1.py
class Solution(object):
    def convert(self, s, numRows):
        """
        :type s: str
        :type numRows: int
        :rtype: str
        """
        
        unitLen = 1 if numRows == 1 else numRows*2 - 2
        rows = [''] * numRows
        
        for i, c in enumerate(s):
            unitIdx = i % unitLen
            rowIdx = unitIdx if unitIdx < numRows else unitLen - unitIdx
            rows[rowIdx] = rows[rowIdx] + c
        
        return "".join(rows)
        

File: 6_ZigZagConversion/6_v1.js
/**
 * @param {string} s
 * @param {number} numRows
 * @return {string}
 */
var convert = function(s, numRows) {
    var unitLen = (numRows == 1)? 1: numRows*2 - 2
    var rows = new Array(numRows).fill('')
    
    for (let i = 0; i < s.length; i++){
        var unitIdx = i % unitLen
        var rowIdx = (unitIdx < numRows)? unitIdx: unitLen - unitIdx
        rows[rowIdx] += s[i]
    }
    
    return rows.join('')
};